### ISSUE TITLE
[Feature Request]: Setting up on mac M series

### ISSUE THREAD

**Issue by dhanush-2501** [unifyai/ivy#23458]
**Is your feature request related to a problem? Please describe.**
 Setting up Dev environment on mac M series devices are really hard. Almost every Mac users are facing this issue. Setting up takes a lot of time and effort but only ends at frustration. Currently most of the Mac users are contributing via codespaces only. But Not everybody can afford codepsace.

**Describe the solution you'd like**
We request an new detailed setting up for Mac M series devices.

**Describe alternatives you've considered**
Currently developing using codespaces but it has time limitations and it costs me.

**Comment by hassaanhameed786** [comment]
yes I just pull the ivy project into my docker I felt this is the right question
 you raised

**Comment by dhanush-2501** [comment]
@hassaanhameed786  A work around as of now is creating a venv with python3.9.
`python3.9 -m venv env_name`

**Comment by hassaanhameed786** [comment]
@Pchatain  I try to do this but after that I didn't because u already told in above

**Comment by Pchatain** [comment]
Yeah the best solution now will be to use a local environment such as venv or conda.

**Comment by dadangsetio** [comment]
@Pchatain i can run ivy on my mac m1, can you help me bro?
